File: app/src/main/java/bj/discogsbrowser/model/collection/CollectionRelease.java

package bj.discogsbrowser.model.collection;

import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;

import java.util.List;

import bj.discogsbrowser.model.common.BasicInformation;
import bj.discogsbrowser.model.common.RecyclerViewModel;
import bj.discogsbrowser.utils.ArtistsBeautifier;

public class CollectionRelease implements RecyclerViewModel
{
    @SerializedName("id")
    @Expose
    private String id;
    @SerializedName("instance_id")
    @Expose
    private String instanceId;
    @SerializedName("folder_id")
    @Expose
    private String folderId;
    @SerializedName("rating")
    @Expose
    private Integer rating;
    @SerializedName("basic_information")
    @Expose
    private BasicInformation basicInformation;
    @SerializedName("notes")
    @Expose
    private List<Note> notes = null;
    private String subtitle;

    public String getId()
    {
        return id;
    }

    public void setId(String id)
    {
        this.id = id;
    }

    public String getInstanceId()
    {
        return instanceId;
    }

    public void setInstanceId(String instanceId)
    {
        this.instanceId = instanceId;
    }

    public String getFolderId()
    {
        return folderId;
    }

    public void setFolderId(String folderId)
    {
        this.folderId = folderId;
    }

    public Integer getRating()
    {
        return rating;
    }

    public void setRating(Integer rating)
    {
        this.rating = rating;
    }

    public BasicInformation getBasicInformation()
    {
        return basicInformation;
    }

    public void setBasicInformation(BasicInformation basicInformation)
    {
        this.basicInformation = basicInformation;
    }

    public List<Note> getNotes()
    {
        return notes;
    }

    public void setNotes(List<Note> notes)
    {
        this.notes = notes;
    }

    @Override
    public String getTitle()
    {
        return basicInformation.getTitle();
    }

    @Override
    public String getSubtitle()
    {
        if (subtitle == null)
        {
            ArtistsBeautifier artistsBeautifier = new ArtistsBeautifier();
            subtitle = artistsBeautifier.formatArtists(basicInformation.getArtists());
        }
        return subtitle;
    }

    @Override
    public String getThumb()
    {
        return basicInformation.getThumb();
    }

    @Override
    public String getType()
    {
        return "release";
    }
}

File: app/src/main/java/bj/vinylbrowser/model/collection/CollectionRelease.kt
package bj.vinylbrowser.model.collection

import android.os.Parcel
import android.os.Parcelable
import bj.vinylbrowser.model.common.BasicInformation
import bj.vinylbrowser.model.common.RecyclerViewModel
import bj.vinylbrowser.utils.ArtistsBeautifier
import com.google.gson.annotations.SerializedName

/**
 * Created by Josh Laird on 19/05/2017.
 */
data class CollectionRelease(@get:JvmName("getId_") var id: String = "",
                             @SerializedName("instance_id") var instanceId: String = "",
                             @SerializedName("folder_id") var folderId: String = "",
                             var rating: Int = 0,
                             @SerializedName("basic_information") var basicInformation: BasicInformation = BasicInformation(),
                             var notes: List<Note> = emptyList(),
                             @get:JvmName("getSubtitle_") var subtitle: String = "")
    : RecyclerViewModel, Parcelable {

    override fun writeToParcel(dest: Parcel, flags: Int) {
        dest.writeString(id)
        dest.writeString(instanceId)
        dest.writeString(folderId)
        dest.writeInt(rating)
        dest.writeParcelable(basicInformation, 0)
        dest.writeList(notes)
        dest.writeString(subtitle)
    }

    override fun describeContents(): Int {
        return 0
    }

    override fun getTitle(): String {
        return basicInformation.title
    }

    override fun getSubtitle(): String {
        if (subtitle == "") {
            val artistsBeautifier = ArtistsBeautifier()
            subtitle = artistsBeautifier.formatArtists(basicInformation.artists)
        }
        return subtitle
    }

    override fun getThumb(): String {
        return basicInformation.thumb
    }

    override fun getType(): String {
        return "release"
    }

    override fun getId(): String {
        return id
    }
}